ear-production-suite/plugins/direct_speakers/CMakeLists.txt

```diff
@@ -10,9 +10,57 @@ set(SOURCES_DIRECT_SPEAKERS_INPUT
     src/direct_speakers_plugin_processor.cpp
 )
 
+set(HEADERS_DIRECT_SPEAKERS_INPUT
+	${CMAKE_CURRENT_SOURCE_DIR}/../../lib/include/detail/constants.hpp
```

Comment:
Could these be added as INTERFACE headers on the library instead
Might mean moving the .cpp files out of the `add_library()` call and into a `target_sources(ear-plugin-base PRIVATE blah_blah.cpp)` as don't think CMake likes you to mix the two forms (with & without visibility specifiers)
e.g.
```cmake
target_sources(ear-plugin-base 
  PRIVATE
  all_the_cpp_files_currently_in_add_library.cpp
  INTERFACE
  include/detail/constants.hpp
  include/direct_speakers_backend.hpp
  include/speaker_setups.hpp
)
```
etc?

If that works in the same way it would mean you'd automatically get them each time you link to the target rather than having to specify for each plugin.

it would also avoid specifying any source files by absolute path which can cause issues (I think, although it might be ok for headers as they more or less get ignored by most of cmake)

You could also set the source_group here so there was one for the library rather than adding them into each plugin.

Don't normally like sticking all the headers into a variable rather than directly into target_sources as it's a bit error prone, but as source_group doesn't take generator expressions it's probably less error-prone than using a regex in source_group. (if it did you could just do `$<TARGET_PROPERTY:ear-plugin-base,INTERFACE_SOURCES>` or something like that)